pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -94,15 +94,26 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 		return nil, err
 	}
 	// Get constrained AMI ID
-	amis, err := p.amiProvider.Get(ctx, instanceTypes)
+	amis, err := p.amiProvider.Get(ctx, constraints, instanceTypes)
 	if err != nil {
 		return nil, err
 	}
 	// Construct launch templates
 	launchTemplates := map[string][]cloudprovider.InstanceType{}
 	for amiID, instanceTypes := range amis {
 		// Get userData for Node
-		userData, err := p.getUserData(ctx, constraints, instanceTypes, additionalLabels)
+		var userData string
+		if constraints.AMIFamily != nil {
+			if strings.EqualFold(*constraints.AMIFamily, v1alpha1.OperatingSystemBottleRocket) {
+				userData, err = p.getBottleRocketUserData(ctx, constraints, additionalLabels)
+			} else if strings.EqualFold(*constraints.AMIFamily, v1alpha1.OperatingSystemEKSOptimized) {
+				userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
+			} else {
+				userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
```

Comment:
If `AMIFamily` is non-empty, but contains an invalid value, shouldn't we at least log something here or maybe even return an error? Or maybe even better add proper validation to [provider_validation.go](https://github.com/aws/karpenter/blob/e6bc0e795412573e3fc1057a07f4680c43a84cca/pkg/cloudprovider/aws/apis/v1alpha1/provider_validation.go)?

Reply:
Added logging here, as well as to the getSSMQuery section in ami.go.